PerfTools/JeProf/src/jeprof.cc

```diff
@@ -0,0 +1,36 @@
+#include "PerfTools/JeProf/interface/jeprof.h"
+#include <string>
+#include <dlfcn.h>
+#include <cstdio>
+#include <cstring>
+
+extern "C" {
+typedef int (*mallctl_t)(const char *name, void *oldp, size_t *oldlenp, void *newp, size_t newlen);
+}
+
+namespace jeprof {
+  bool initialize_prof();
+
+  mallctl_t mallctl = nullptr;
+  const bool have_jemalloc_and_prof = initialize_prof();
+
+  bool initialize_prof() {
+    // check if mallctl and friends are available, if we are using jemalloc
+    mallctl = (mallctl_t)::dlsym(RTLD_DEFAULT, "mallctl");
+    if (mallctl == nullptr)
+      return false;
+    // check if heap profiling available, if --enable-prof was specified at build time
+    bool enable_prof = false;
+    size_t bool_s = sizeof(bool);
+    mallctl("prof.active", &enable_prof, &bool_s, nullptr, 0);
+    return enable_prof;
+  }
```

Comment:
I'd suggest to move everything else than `makeHeapDump()` function to anonymous namespace instead, to make the global variables and other symbols inaccessible from outside of this file

Reply:
OK